ee/session_recordings/playlist_counters/recordings_that_match_playlist_filters.py

```diff
@@ -302,38 +384,45 @@ def count_recordings_that_match_playlist_filters(playlist_id: int) -> None:
             playlist = SessionRecordingPlaylist.objects.get(id=playlist_id)
             redis_client = get_client()
 
-            existing_value = redis_client.get(f"{PLAYLIST_COUNT_REDIS_PREFIX}{playlist.short_id}")
+            existing_value = redis_client.getex(
+                name=f"{PLAYLIST_COUNT_REDIS_PREFIX}{playlist.short_id}", ex=THIRTY_SIX_HOURS_IN_SECONDS
+            )
             if existing_value:
                 existing_value = json.loads(existing_value)
             else:
                 existing_value = {}
 
-            # if we have results from the last hour we don't need to run the query
-            if existing_value.get("refreshed_at"):
-                last_refreshed_at = datetime.fromisoformat(existing_value["refreshed_at"])
-                seconds_since_refresh = int((datetime.now() - last_refreshed_at).total_seconds())
-
-                if seconds_since_refresh <= settings.PLAYLIST_COUNTER_PROCESSING_COOLDOWN_SECONDS:
-                    REPLAY_TEAM_PLAYLIST_COUNT_SKIPPED.labels(reason="cooldown").inc()
-                    return
-
-            # if this is the default filters, then we shouldn't have allowed this to be created - we can skip it
-            if playlist.filters == DEFAULT_RECORDING_FILTERS:
-                REPLAY_TEAM_PLAYLIST_COUNT_SKIPPED.labels(reason="default_filters").inc()
+            if should_skip_task(existing_value, playlist.filters):
                 return
 
             query = convert_filters_to_recordings_query(playlist)
+
+            # if we already have some data and the query is sorted by start_time,
+            # we can query only new recordings, to (hopefully) reduce load on CH
+            has_existing_data = existing_value.get("refreshed_at", None)
+            can_query_only_new_recordings = query.order == "start_time"
+
+            if has_existing_data and can_query_only_new_recordings:
+                query.date_from = existing_value["refreshed_at"]
```

Comment:
check if we can optimise by only loading new data

Reply:
this is likely the majority case

in future we _could_ check things like whether `date_to` is set but there are essentially zero of these right now